core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -343,8 +343,8 @@ public void commit() {
         LOG.warn("Failed to load committed snapshot, skipping manifest clean-up");
       }
 
-    } catch (RuntimeException e) {
-      LOG.warn("Failed to load committed table metadata, skipping manifest clean-up", e);
+    } catch (Throwable t) {
```

Comment:
My understanding has always been that we should only catch non-fatal exceptions.

Is there anyway we can slim down the scope / amount of code that this larger `try-catch` block covers?